tsdb/meta.go

```diff
@@ -1360,11 +1349,16 @@ func (s *Series) UnmarshalBinary(buf []byte) error {
 
 // InitializeShards initializes the list of shards.
 func (s *Series) InitializeShards() {
+	s.mu.Lock()
+	defer s.mu.Unlock()
 	s.shardIDs = make(map[uint64]bool)
 }
 
 // match returns true if all tags match the series' tags.
 func (s *Series) match(tags map[string]string) bool {
```

Comment:
I can't find where this function is called, and since it's non-exported, we should just remove it.